File: frontend/src/vista-registrarse.ts
import { LitElement, html, css, customElement } from 'lit-element';
import '@vaadin/vaadin-text-field/src/vaadin-text-field.js';
import '@vaadin/vaadin-text-field/src/vaadin-password-field.js';
import '@vaadin/vaadin-ordered-layout/src/vaadin-vertical-layout.js';
import '@vaadin/vaadin-ordered-layout/src/vaadin-horizontal-layout.js';
import '@vaadin/vaadin-button/src/vaadin-button.js';

@customElement('vista-registrarse')
export class VistaRegistrarse extends LitElement {
  static get styles() {
    return css`
      :host {
          display: block;
          height: 100%;
      }
      `;
  }

  render() {
    return html`
<vaadin-vertical-layout style="width: 100%; align-items: center; padding-right: var(--lumo-space-m); padding-bottom: var(--lumo-space-m); padding-left: var(--lumo-space-m); position: absolute; justify-content: space-between;" theme="spacing-l">
 <vaadin-horizontal-layout id="Layoutregistrarse" style="width: 100%; flex-shrink: 0; justify-content: center;">
  <label style="flex-shrink: 0; font-size: 400%; flex-grow: 0;" id="Labelregistrarse">Registrarse</label>
 </vaadin-horizontal-layout>
 <vaadin-vertical-layout style="flex-shrink: 0; align-self: center; width: 70%; flex-grow: 0; justify-content: space-between; height: 75%;" theme="spacing">
  <vaadin-vertical-layout style="width: 100%; flex-shrink: 0; justify-content: space-between;" theme="spacing-xs">
   <vaadin-horizontal-layout style="width: 100%; flex-shrink: 0;" id="Layoutemailregistrarse">
    <label style="width: 20%; flex-shrink: 0; align-self: center; padding-right: var(--lumo-space-xl); flex-grow: 0;" id="Labelemailregistrarse">Email</label>
    <vaadin-text-field placeholder="Email" style="flex-shrink: 0; flex-grow: 1;" id="Textfieldemailregistrarse"></vaadin-text-field>
   </vaadin-horizontal-layout>
   <vaadin-horizontal-layout style="width: 100%; flex-shrink: 0;" id="Layoutnombreregistrarse">
    <label style="width: 20%; flex-shrink: 0; align-self: center; padding-right: var(--lumo-space-xl);" id="Labelnombreregistrarse">Nombre</label>
    <vaadin-text-field placeholder="Nombre" style="flex-shrink: 0; flex-grow: 1;" id="Textfieldnombreregistrarse"></vaadin-text-field>
   </vaadin-horizontal-layout>
   <vaadin-horizontal-layout style="width: 100%; flex-shrink: 0;" id="Layoutapellidosregistrarse">
    <label style="width: 20%; flex-shrink: 0; align-self: center; flex-grow: 0; padding-right: var(--lumo-space-xl);" id="Labelapellidosregistrarse">Apellidos</label>
    <vaadin-text-field placeholder="Apellidos" style="flex-shrink: 0; flex-grow: 1;" id="Textfieldapellidosregistrarse"></vaadin-text-field>
   </vaadin-horizontal-layout>
   <vaadin-horizontal-layout style="width: 100%; flex-shrink: 0;" id="Layouttelefonoregistrarse">
    <label style="width: 20%; flex-shrink: 0; align-self: center; flex-grow: 0; padding-right: var(--lumo-space-xl);" id="Labeltelefonoregistrarse">Teléfono</label>
    <vaadin-text-field placeholder="Teléfono" style="flex-shrink: 0; flex-grow: 1;" id="Textfieldtelefonoregistrarse"></vaadin-text-field>
   </vaadin-horizontal-layout>
   <vaadin-horizontal-layout style="width: 100%; flex-shrink: 0;" id="Layoutfechanacimientoregistrarse">
    <label style="width: 20%; flex-shrink: 0; align-self: center; flex-grow: 0; padding-right: var(--lumo-space-xl);" id="Labelfechanacimientoregistrarse">Fecha Nacimiento</label>
    <vaadin-text-field placeholder="Fecha Nacimiento (DD-MM-AAAA)" style="flex-shrink: 0; align-self: center; flex-grow: 1;" id="Textfieldfechanacimientoregistrarse"></vaadin-text-field>
   </vaadin-horizontal-layout>
   <vaadin-horizontal-layout style="width: 100%; flex-shrink: 0;" id="Layoutapodoregistrarse">
    <label style="width: 20%; flex-shrink: 0; align-self: center; flex-grow: 0; padding-right: var(--lumo-space-xl);" id="Labelapodoregistrarse">Nick</label>
    <vaadin-text-field placeholder="Nick" style="flex-shrink: 0; flex-grow: 1;" id="Textfieldregistrarse"></vaadin-text-field>
   </vaadin-horizontal-layout>
   <vaadin-horizontal-layout style="width: 100%; flex-shrink: 0;" id="Layoutdniregistrarse">
    <label style="width: 20%; flex-shrink: 0; align-self: center; padding-right: var(--lumo-space-xl);" id="Labeldniregistrarse">DNI</label>
    <vaadin-text-field placeholder="DNI" style="flex-shrink: 0; flex-grow: 1;" id="Textfielddniregistrarse"></vaadin-text-field>
   </vaadin-horizontal-layout>
   <vaadin-horizontal-layout style="width: 100%; flex-shrink: 0;" id="Layoutntarjetaregistrarse">
    <label style="width: 20%; flex-shrink: 0; align-self: center; padding-right: var(--lumo-space-xl);" id="Labelntarjetaregistrarse">Nº Tarjeta</label>
    <vaadin-text-field placeholder="Nº Tarjeta" style="flex-shrink: 0; flex-grow: 1;" id="Textfieldntarjetaregistrarse"></vaadin-text-field>
   </vaadin-horizontal-layout>
   <vaadin-horizontal-layout style="width: 100%;" id="Layoutcontrasenaregistrarse">
    <label style="width: 20%; flex-shrink: 0; align-self: center; padding-right: var(--lumo-space-xl);" id="Labelcontrasenaregistrarse">Contraseña</label>
    <vaadin-password-field placeholder="Contraseña" style="flex-shrink: 0; flex-grow: 1;" has-value id="Passwordfieldcontrasenaregistrarse"></vaadin-password-field>
   </vaadin-horizontal-layout>
   <vaadin-horizontal-layout style="width: 100%;" id="Layoutrepetircontrasenaregistrarse">
    <label style="width: 20%; flex-shrink: 0; align-self: center; padding-right: var(--lumo-space-xl);" id="Labelrepetircontrasenaregistrarse">Repetir Contraseña</label>
    <vaadin-password-field placeholder="Repetir Contraseña" style="flex-shrink: 0; align-self: center; flex-grow: 1;" has-value id="Passwordfieldrepetircontrasenaregistrarse"></vaadin-password-field>
   </vaadin-horizontal-layout>
   <label style="flex-shrink: 0; font-size: 120%; width: 100%; align-self: center; font-size: 100%; text-align: center; color: #FF0000;" id="Labelerrordatosregistrarse"></label>
  </vaadin-vertical-layout>
  <vaadin-horizontal-layout style="flex-shrink: 0; width: 80%; justify-content: space-around; align-self: center; padding-top: var(--lumo-space-s); padding-bottom: var(--lumo-space-m);" id="Layoutcancelarenviarregistrarse">
   <vaadin-button style="flex-shrink: 0; align-self: center; width: 40%;" id="Cancelarregistrarse">
     Cancelar 
   </vaadin-button>
   <vaadin-button style="flex-shrink: 0; align-self: center; width: 40%;" id="Enviarregistrarse">
     Enviar 
   </vaadin-button>
  </vaadin-horizontal-layout>
 </vaadin-vertical-layout>
</vaadin-vertical-layout>
`;
  }

  // Remove this method to render the contents of this view inside Shadow DOM
  createRenderRoot() {
    return this;
  }
}

File: src/main/java/vistas/VistaRegistrarse.java
package vistas;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.littemplate.LitTemplate;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.component.template.Id;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.html.Image;
import com.vaadin.flow.component.html.Label;
import com.vaadin.flow.component.textfield.TextField;
import com.vaadin.flow.component.textfield.PasswordField;

@Tag("vista-registrarse")
@JsModule("./src/vista-registrarse.ts")
public class VistaRegistrarse extends LitTemplate {


	@Id("Layoutregistrarse")
	private HorizontalLayout layoutregistrarse;
	@Id("Logoregistrarse")
	private Button logoregistrarse;
	@Id("Imagenlogoregistrarse")
	private Image imagenlogoregistrarse;
	@Id("Labelregistrarse")
	private Label labelregistrarse;
	@Id("Layoutemailregistrarse")
	private HorizontalLayout layoutemailregistrarse;
	@Id("Labelemailregistrarse")
	private Label labelemailregistrarse;
	@Id("Textfieldemailregistrarse")
	private TextField textfieldemailregistrarse;
	@Id("Layoutnombreregistrarse")
	private HorizontalLayout layoutnombreregistrarse;
	@Id("Labelnombreregistrarse")
	private Label labelnombreregistrarse;
	@Id("Textfieldnombreregistrarse")
	private TextField textfieldnombreregistrarse;
	@Id("Layoutapellidosregistrarse")
	private HorizontalLayout layoutapellidosregistrarse;
	@Id("Labelapellidosregistrarse")
	private Label labelapellidosregistrarse;
	@Id("Textfieldapellidosregistrarse")
	private TextField textfieldapellidosregistrarse;
	@Id("Layoutfechanacimientoregistrarse")
	private HorizontalLayout layoutfechanacimientoregistrarse;
	@Id("Labelfechanacimientoregistrarse")
	private Label labelfechanacimientoregistrarse;
	@Id("Textfieldfechanacimientoregistrarse")
	private TextField textfieldfechanacimientoregistrarse;
	@Id("Layoutapodoregistrarse")
	private HorizontalLayout layoutapodoregistrarse;
	@Id("Labelapodoregistrarse")
	private Label labelapodoregistrarse;
	@Id("Textfieldregistrarse")
	private TextField textfieldregistrarse;
	@Id("Layoutdniregistrarse")
	private HorizontalLayout layoutdniregistrarse;
	@Id("Labeldniregistrarse")
	private Label labeldniregistrarse;
	@Id("Textfielddniregistrarse")
	private TextField textfielddniregistrarse;
	@Id("Layoutntarjetaregistrarse")
	private HorizontalLayout layoutntarjetaregistrarse;
	@Id("Labelntarjetaregistrarse")
	private Label labelntarjetaregistrarse;
	@Id("Textfieldntarjetaregistrarse")
	private TextField textfieldntarjetaregistrarse;
	@Id("Layoutcontrasenaregistrarse")
	private HorizontalLayout layoutcontrasenaregistrarse;
	@Id("Labelcontrasenaregistrarse")
	private Label labelcontrasenaregistrarse;
	@Id("Passwordfieldcontrasenaregistrarse")
	private PasswordField passwordfieldcontrasenaregistrarse;
	@Id("Layoutrepetircontrasenaregistrarse")
	private HorizontalLayout layoutrepetircontrasenaregistrarse;
	@Id("Labelrepetircontrasenaregistrarse")
	private Label labelrepetircontrasenaregistrarse;
	@Id("Passwordfieldrepetircontrasenaregistrarse")
	private PasswordField passwordfieldrepetircontrasenaregistrarse;
	@Id("Layoutcancelarenviarregistrarse")
	private HorizontalLayout layoutcancelarenviarregistrarse;
	@Id("Cancelarregistrarse")
	private Button cancelarregistrarse;
	@Id("Enviarregistrarse")
	private Button enviarregistrarse;

	public VistaRegistrarse() {
        // You can initialise any data required for the connected UI components here.
    }

}